### PR TITLE
Fix passwords to use all characters

### LINKED ISSUE #267: SQLi vulnerability in user creation
https://github.com/kartoza/docker-postgis/blob/18f99bc8d65bba72f4f5fe7f63d9e0fa179c73b2/scripts/setup-user.sh#L28

If you try to create a user with a password like `' or 1=1; DROP TABLE pg_roles --` you are going to have a bad day.

### PR DESCRIPTION
**Pull request by NyakudyaA**
Fixes #267 

* Initially tried to use sed to do escape all special characters but realised that they were passed to postgres as is there invalidating the user credentials.

@lucernae Can you test

### CODE REVIEW

File: scripts/setup-user.sh [hunk 1]
```diff
@@ -14,24 +14,44 @@ source /scripts/env-data.sh
...
 if [ -z "$RESULT" ]; then
 	COMMAND="CREATE"
 fi
-su - postgres -c "psql postgres -c \"$COMMAND USER $POSTGRES_USER WITH SUPERUSER ENCRYPTED PASSWORD '$POSTGRES_PASS';\""
 
-echo "Creating replication user $REPLICATION_USER"
+echo "Creating superuser user $POSTGRES_USER using $PASSWORD_AUTHENTICATION authentication "
+if [ PASSWORD_AUTHENTICATION="md5" ]; then
+	PG_PASS=$(U=$POSTGRES_USER; P=$(cat /tmp/PGPASSWORD.txt); echo -n md5; echo -n $P$U | md5sum | cut -d' ' -f1)
+	su - postgres -c "psql postgres -c \"$COMMAND USER $POSTGRES_USER WITH SUPERUSER  PASSWORD '$PG_PASS';\""
```
Comment (lucernae) [inline scripts/setup-user.sh:29]: I don't understand this two line.
You append password and user together using `echo -n $P$U` then pass it to md5sum and extract the digest.
This digest is then used as password.
Wouldn't that make the digest the password instead?

Comment (NyakudyaA) [reply scripts/setup-user.sh:29]: @lucernae  
When you use use 
```
CREATE USER docker WITH SUPERUSER ENCRYPTED PASSWORD 'foo';
```
 The encryption algorithm that is used for the password will depend on the cluster initialization parameters. In this case we only using md5 or sha256.
If you pass an encrypted password Postgres is smart to figure out that the password is already encrypted and will store it as.
To test try the following 
```
CREATE USER kartoza with password `md5a847d0f7cda274f87440e9e9a3312394`;
``` 
Then log in using the credentials user=`kartoza`;pass=`kartoza`

**Comment by NyakudyaA** [comment]
Ping @lucernae Should we merge here ?

File: scripts/setup-user.sh (hunk 1 above)
Comment (lucernae) [reply scripts/setup-user.sh:29]: ok, noted I will test it.

**Comment by lucernae** [comment]
Oh yeah, sorry @NyakudyaA forgot to test it again.

**Pull request merged**